tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Framework.cs

```diff
@@ -468,7 +468,7 @@ public Dictionary<string, object> GetBasicRequestArgsForWaf()
 
     private static object GetHeaderAsArray(string[] value) => value.Length == 1 ? value[0] : value;
 
-    private static object GetHeaderValueForWaf(NameValueCollection headers, string currentKey) => GetHeaderAsArray(headers.GetValues(currentKey));
+    private static object GetHeaderValueForWaf(NameValueCollection headers, string currentKey) => GetHeaderAsArray(headers.GetValues(currentKey) ?? [string.Empty]);
```

Comment:
Do you _really_ want to return `string.Empty` in the array here? 🤔 This will allocate a new array every time.

Reply:
true, would an empty array be better?

Reply:
Yes, the `[]` empty array will re-use the static `Array.Empty<T>`, so no extra allocations 🙂

Reply:
thanks!  fixed in [cddfd73](https://github.com/DataDog/dd-trace-dotnet/pull/6211/commits/cddfd730ba32e1f632ceaa33a5d1315006f08fb9)

Reply:
actually tests arent passing anymore :( gotta find out why

Reply:
so it's all normal, it s just that now that cookies dictionary are empty, we dont send them to the waf, which is better for performances (no unnecessary encoding). So I'm updating all snapshots accordingly, all fingerprint for session which were just saying "empty session" are removed